Source/ALSV4_CPP/Public/Character/ALSBaseCharacter.h

```diff
@@ -625,4 +627,7 @@ class ALSV4_CPP_API AALSBaseCharacter : public ACharacter
 
 	/** We won't use curve based movement and a few other features on networked games */
 	bool bEnableNetworkOptimizations = false;
+
+private:
+	UALSDebugComponent* m_debugComponent = nullptr;
```

Comment:
Debug component was designed to be a plug-in/plug-out component. Thus, I think it shouldn't be referenced in character class by design.

Any chance we do not reference debug component in character class, and self contamine everything in component class instead?

Reply:
I was also thinking a lot about this dependency and honestly I also don't like it! What would be needed is a kind of event that is triggered if the traces should be drawn in every consumer like ALSBaseCharacter, ALSPlayerCameraManager, ALSCharacterAnimInstance and ALSMantleComponent.

Do you think it would make sense to provide a bool variable in in ALSBaseCharacter, which could be returned by a getter to indicate in the system that the traces should be drawn? Additionally, it could make sense to forward the state in ALSDebugComponent to the ALSBaseCharacter. I really like the idea with the getter method as it would avoid referencing the debug component in every comsumer class. Moreover, the debug draw functions could also be moved to a new static class like ALSMathLibrary.
But what I don't like with this approach that someone could enable the debug traces from the public API of  ALSBaseCharacter. If you want we could also have a short discord conversion to find a good solution for this issue.